client.go

```diff
@@ -3,64 +3,70 @@ package supabase
 import (
 	"errors"
 
+	"github.com/supabase-community/gotrue-go"
+	"github.com/supabase-community/gotrue-go/types"
+	postgrest "github.com/supabase-community/postgrest-go"
 	storage_go "github.com/supabase-community/storage-go"
-	"github.com/supabase/postgrest-go"
 )
 
 const (
 	REST_URL     = "/rest/v1"
 	STORGAGE_URL = "/storage/v1"
+	AUTH_URL     = "/auth/v1"
 )
 
 type Client struct {
-	rest    *postgrest.Client
-	Storage *storage_go.Client
+	// Why is this a private field??
+	rest    postgrest.Client
+	Storage storage_go.Client
+	Auth    gotrue.Client
+	options clientOptions
 }
 
-type RestOptions struct {
-	Schema string
-}
-
-type ClientOptions struct {
-	Headers map[string]string
-	Db      *RestOptions
+type clientOptions struct {
+	url     string
+	headers map[string]string
 }
 
 // NewClient creates a new Supabase client.
 // url is the Supabase URL.
 // key is the Supabase API key.
 // options is the Supabase client options.
-func NewClient(url, key string, options *ClientOptions) (*Client, error) {
+func NewClient(url, key string, schema string, headers map[string]string) (*Client, error) {
 	if url == "" || key == "" {
 		return nil, errors.New("url and key are required")
 	}
 
-	headers := map[string]string{
-		"Authorization": "Bearer " + key,
-		"apikey":        key,
+	if headers == nil {
+		headers = map[string]string{}
 	}
 
-	if options != nil && options.Headers != nil {
-		for k, v := range options.Headers {
+	headers["Authorization"] = "Bearer " + key
+	headers["apikey"] = key
+
+	if headers != nil {
+		for k, v := range headers {
 			headers[k] = v
 		}
 	}
 
-	var schema string
-	if options != nil && options.Db != nil && options.Db.Schema != "" {
-		schema = options.Db.Schema
-	} else {
+	client := &Client{}
+	client.options.url = url
+	// map is pass by reference, so this gets updated by rest of function
+	client.options.headers = headers
+
+	if schema == "" {
 		schema = "public"
 	}
-	if options != nil && options.Headers != nil {
```

Comment:
This was redundant. If a user had passed in headers they would've been added twice.

Reply:
Thanks, let remove it.